src/Backend/Core/Ajax/GenerateUrl.php

```diff
@@ -22,10 +22,10 @@ public function execute(): void
         parent::execute();
 
         // get parameters
-        $url = \SpoonFilter::getPostValue('url', null, '', 'string');
-        $className = \SpoonFilter::getPostValue('className', null, '', 'string');
-        $methodName = \SpoonFilter::getPostValue('methodName', null, '', 'string');
-        $parameters = \SpoonFilter::getPostValue('parameters', null, '', 'string');
+        $url = $this->getRequest()->request->get('url', '');
```

Comment:
Don't think much can be changed here, as `$this->get('fork.repository.meta')->generateURL` expects strings